vitrina/datasets/views.py

```diff
@@ -1488,7 +1489,12 @@ def form_valid(self, form):
         self.object.dataset.current_structure = self.object
         self.object.dataset.save()
         set_comment(_(f'Added Structure file "{self.object.file}".'))
-        create_structure_objects(self.object)
+
+        if self.metadata_version and self.metadata_version.status != VersionStatus.DRAFT:
```

Comment:
As this view has mandatory url argument `version_id` it must be `Version` entry in db.  If you provide `version_id` and cannot find `Version` - this is inconsistent data and we need to stop import here by raising a `400` error.
I suggest to modify `DatasetStructureMixin` or create  another and raise error there.